src/traces/histogram/defaults.js

```diff
@@ -22,8 +22,10 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
     }
 
-    var x = coerce('x'),
-        y = coerce('y');
+    var x = coerce('x');
+    var y = coerce('y');
+    var hasX = x && x.length;
```

Comment:
I suspect `x && x.length` is faster than `Array.isArray(x) && x.length`. Might be worth standardizing `lib/is_array.js` (as e.g. `Lib.isNonEmptyArray`) ?

Reply:
> e.g. `Lib.isNonEmptyArray`

Ah hmm, before I do that, I want to check this against our previous behavior for trace types that don't necessarily need both coordinates specified - like scatter, which can use `x0`/`dx` along with a `y` array. For example if `x=[], y=[1, 2, 3]` we should treat that as `_length=0`, not `3`, right? I'll need to look over our test coverage for those cases...